### PR TITLE
Add Kay60 PCB

### PR DESCRIPTION
**Pull request by mechlovin**
## Description

Add Kay60 PCB, Atmega32u4, RGB underglow.
Thank you.
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [ ] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by drashna** [comment]
This has changes to the submodules which need to be reverted, first. 

```sh
git checkout master
git submodule foreach 'git log -n1 --oneline'
# remember the first abcdef1234 on the submodules that have an issue
git checkout @@your-branch@@
# for each of the problematic repo's:
cd lib/@@name@@
git checkout abcdef1234
```

File: keyboards/mechlovin/kay60/rules.mk [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+#
+BOOTMAGIC_ENABLE = yes      # Enable Bootmagic Lite
+MOUSEKEY_ENABLE = yes       # Mouse keys
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = no         # Console for debug
+COMMAND_ENABLE = no         # Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = yes            # USB Nkey Rollover
```
Comment (fauxpark) [inline keyboards/mechlovin/kay60/rules.mk:18]:
```suggestion
NKRO_ENABLE = yes            # Enable N-Key Rollover
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Comment by mechlovin** [comment]
> This has changes to the submodules which need to be reverted, first.
> 
> ```shell
> git checkout master
> git submodule foreach 'git log -n1 --oneline'
> # remember the first abcdef1234 on the submodules that have an issue
> git checkout @@your-branch@@
> # for each of the problematic repo's:
> cd lib/@@name@@
> git checkout abcdef1234
> ```

I followed your instruction and there seem to be no changes recorded in the current repo. Please take a look at the attached screenshot and let me know if I need to do anything else. Thank you!

**Comment by mechlovin** [comment]
> > This has changes to the submodules which need to be reverted, first.
> > ```shell
> > git checkout master
> > git submodule foreach 'git log -n1 --oneline'
> > # remember the first abcdef1234 on the submodules that have an issue
> > git checkout @@your-branch@@
> > # for each of the problematic repo's:
> > cd lib/@@name@@
> > git checkout abcdef1234
> > ```
> 
> I followed your instruction and there seem to be no changes recorded in the current repo. Please take a look at the attached screenshot and let me know if I need to do anything else. Thank you!

Sorry I forgot to attach the screenshot. 
Kay60

**Comment by drashna** [comment]
If you want, I can see about fixing the branch.  Though, you'd need to use `git reset` locally, to pull the changes afterwards.

**Comment by noroadsleft** [comment]
If you have QMK's `qmk_firmware` repository configured as `upstream` in Git:

```
git checkout kay60
git fetch upstream master
git rebase upstream/master
git push --force-with-lease origin HEAD:kay60
```

**Review by fauxpark — APPROVED** [review]

**Review by drashna — APPROVED** [review]

**Comment by drashna** [comment]
Thanks!

**Pull request merged**